sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/implementation/AssistantThreadsImpl.java

```diff
@@ -58,7 +58,7 @@ public final class AssistantThreadsImpl {
      * The interface defining all the services for AssistantsClientAssistantThreads to be used by the proxy service to
      * perform REST calls.
      */
-    @Host("{endpoint}/openai")
+    @Host("{endpoint}")
```

Comment:
TypeSpec Discussion: If we can make the changes from TypeSpec definition or emitter to adopt this `@Host("{endpoint}")`. It will be flexible and work for both Azure and Non-Azure endpoints. 

Also see changes in [this](https://github.com/Azure/azure-sdk-for-java/pull/38282/files#diff-5d92a1c5b5fa0c2352299a33825cc485a6d85f01bb4e439bbafb9102135d5726R275):

```
AssistantsClientImpl client
            = new AssistantsClientImpl(localPipeline, JacksonAdapter.createDefaultSerializerAdapter(),
                useNonAzureOpenAIService() ? OPEN_AI_ENDPOINT : this.endpoint + "/openai");
```